addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2062,10 +2062,26 @@ const ListUserValueWidget = UserValueWidget.extend({
             trEl.appendChild(checkboxTdEl);
         }
         const buttonTdEl = document.createElement('td');
-        buttonTdEl.appendChild(buttonEl);
+        if (!recordData || !recordData.undeletable) {
+            buttonTdEl.appendChild(buttonEl);
+        }
         trEl.appendChild(buttonTdEl);
         this.listTable.appendChild(trEl);
     },
+    /**
+     * Generates a string ID.
+     * 
+     * @private
+     * @param  {Int} length the length of the generated ID.
+     */
+    _generateId(length) {
+        let result = '';
+        const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
```

Comment:
There should definitely be an already existing util method somewhere in Odoo for this, to quickly check

Reply:
No there is not, there is a private function in option.js of website_form. If you want I can make a utils function and use it here (and in website_form)

Reply:
I think it will be better to place it inside an util file than just this widget indeed :+1: 
I'd name it more explicitly tho, you don't know what type of ID generateId will provide.
Maybe something like `_generateHTMLId()` or `_generateDOMId()` or anything you come up with :)